schedule_data.py

```diff
@@ -0,0 +1,280 @@
+import pickle
+import logging
+from objects import Session, Day, User
+from pathlib import Path
+
+root = Path(".")
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(format="%(message)s", level=logging.DEBUG)
+
+formatter = logging.Formatter("%(asctime)s:%(name)s:%(funcName)s:%(levelname)s:%(message)s")
+
+file_handler = logging.FileHandler("logs.txt")
+file_handler.setFormatter(formatter)
+file_handler.setLevel(logging.INFO)
+
+logger.addHandler(file_handler)
+
+
+class ScheduleDataService:
+
+    def create_object_path(self, active_user):
+
+        users_object_path = root / "user_objects" / active_user
+
+        return users_object_path
+
+    def save_users_schedule(self, users_schedule, users_object_path, active_user):
+        self._ensure_database_exists(active_user)
+
+        schedule = open(users_object_path, "wb")
+        pickle.dump(users_schedule, schedule)
+        schedule.close()
+
+        logger.debug("\nYour schedule has been successfully saved \n")
+        logger.info(active_user + "has successfully saved their schedule.")
+
+    def load_users_schedule(self, active_user):
+        self._ensure_database_exists(active_user)
+
+        users_object_path = self.create_object_path(active_user)
+        schedule = open(users_object_path, "rb")
+        users_schedule = pickle.load(schedule)
+        schedule.close()
+
+        return users_schedule
+
+    def _ensure_database_exists(self, active_user):
+        users_object_path = self.create_object_path(active_user)
+        if self._schedule_database_exists(users_object_path):
+            return
+        self._create_user_database(users_object_path)
+
+    def _schedule_database_exists(self, users_object_path):
+
+        return Path(users_object_path).is_file()
+
+    def _create_user_database(self, users_object_path):
+        try:
+            users = open(users_object_path, "wb")
+            pickle.dump({}, users)
+        except Exception as error:
+            logger.error("database creation failed: " + error)
+            raise ScheduleDataException("Sorry but database creation has failed.")
+
+
+class CreateNewSchedule:
+
+    _schedule_data_service: ScheduleDataService
+
+    def __init__(self, schedule_data_service):
+        self._schedule_data_service = schedule_data_service
+
+    def add_sessions(self, active_user):
```

Comment:
I moved the methods into the new class you suggested and tested to make sure they still work when called. 

I also noticed there is a similar class in monthly_variables that takes user input. All methods in this class do, so that isn't the issue, but I was hoping to get your feedback on how it's named.